### PR TITLE
feat: popover for tickets sent to others button on my ticket screen.

### PR DESCRIPTION
**Pull request by reidzeibel**
closes https://github.com/AtB-AS/kundevendt/issues/16927

This will show a popover to let the user knows where to find their sent tickets.
Added some comments to explain what the code means.

####  yarn test -t isViewVisibleOnScreen --verbose   
```
  isViewVisibleOnScreen: view size <= window size, view located on top left (0,0)
    ✓ window size 10 x 10, view size 0 x 0 located top left (0,0) (4 ms)
    ✓ window size 10 x 10, view size 1 x 1 located top left (0,0)
    ✓ window size 10 x 10, view size 2 x 2 located top left (0,0)
    ✓ window size 10 x 10, view size 3 x 3 located top left (0,0)
    ✓ window size 10 x 10, view size 4 x 4 located top left (0,0) (1 ms)
    ✓ window size 10 x 10, view size 5 x 5 located top left (0,0)
    ✓ window size 10 x 10, view size 6 x 6 located top left (0,0)
    ✓ window size 10 x 10, view size 7 x 7 located top left (0,0)
    ✓ window size 10 x 10, view size 8 x 8 located top left (0,0)
    ✓ window size 10 x 10, view size 9 x 9 located top left (0,0)
    ✓ window size 10 x 10, view size 10 x 10 located top left (0,0)
  isViewVisibleOnScreen: view size > window size, view located on top left (0,0)
    ✓ window size 10 x 10, view size 11 x 11 located top left (0,0)
    ✓ window size 10 x 10, view size 12 x 12 located top left (0,0)
    ✓ window size 10 x 10, view size 13 x 13 located top left (0,0)
    ✓ window size 10 x 10, view size 14 x 14 located top left (0,0)
    ✓ window size 10 x 10, view size 15 x 15 located top left (0,0)
    ✓ window size 10 x 10, view size 16 x 16 located top left (0,0)
    ✓ window size 10 x 10, view size 17 x 17 located top left (0,0)
    ✓ window size 10 x 10, view size 18 x 18 located top left (0,0)
    ✓ window size 10 x 10, view size 19 x 19 located top left (0,0)
    ✓ window size 10 x 10, view size 20 x 20 located top left (0,0)
  isViewVisibleOnScreen: view size <= window size, view positioned at center of the window (5,5)
    ✓ window size 10 x 10, view size 0 x 0 located center (5,5)
    ✓ window size 10 x 10, view size 1 x 1 located center (5,5)
    ✓ window size 10 x 10, view size 2 x 2 located center (5,5)
    ✓ window size 10 x 10, view size 3 x 3 located center (5,5)
    ✓ window size 10 x 10, view size 4 x 4 located center (5,5)
    ✓ window size 10 x 10, view size 5 x 5 located center (5,5) (1 ms)
    ✓ window size 10 x 10, view size 6 x 6 located center (5,5)
    ✓ window size 10 x 10, view size 7 x 7 located center (5,5)
    ✓ window size 10 x 10, view size 8 x 8 located center (5,5)
    ✓ window size 10 x 10, view size 9 x 9 located center (5,5)
    ✓ window size 10 x 10, view size 10 x 10 located center (5,5)
  isViewVisibleOnScreen: view located below the window
    ✓ window size 10 x 10, view size 5 x 5 located (11,11)
    ✓ window size 10 x 10, view size 5 x 5 located (12,12)
    ✓ window size 10 x 10, view size 5 x 5 located (13,13)
    ✓ window size 10 x 10, view size 5 x 5 located (14,14)
    ✓ window size 10 x 10, view size 5 x 5 located (15,15)
    ✓ window size 10 x 10, view size 5 x 5 located (16,16)
    ✓ window size 10 x 10, view size 5 x 5 located (17,17) (1 ms)
    ✓ window size 10 x 10, view size 5 x 5 located (18,18)
    ✓ window size 10 x 10, view size 5 x 5 located (19,19)
    ✓ window size 10 x 10, view size 5 x 5 located (20,20)
  isViewVisibleOnScreen: located at negative coordinates
    ✓ window size 10 x 10, view size 5 x 5, located at (-1, -1) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-2, -2) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-3, -3) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-4, -4) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-5, -5) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-6, -6) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-7, -7) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-8, -8) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-9, -9) 
    ✓ window size 10 x 10, view size 5 x 5, located at (-10, -10) 
  isViewVisibleOnScreen: negative size 
    ✓ window size 10 x 10, view size -1 x -1 located center (5,5)
    ✓ window size 10 x 10, view size -2 x -2 located center (5,5)
    ✓ window size 10 x 10, view size -3 x -3 located center (5,5)
    ✓ window size 10 x 10, view size -4 x -4 located center (5,5) (1 ms)
    ✓ window size 10 x 10, view size -5 x -5 located center (5,5)
    ✓ window size 10 x 10, view size -6 x -6 located center (5,5)
    ✓ window size 10 x 10, view size -7 x -7 located center (5,5)
    ✓ window size 10 x 10, view size -8 x -8 located center (5,5)
    ✓ window size 10 x 10, view size -9 x -9 located center (5,5)
    ✓ window size 10 x 10, view size -10 x -10 located center (5,5)
  isViewVisibleOnScreen: negative size and position
    ✓ window size 10 x 10, view size -1 x -1 located (-1,-1)
    ✓ window size 10 x 10, view size -2 x -2 located (-2,-2)
    ✓ window size 10 x 10, view size -3 x -3 located (-3,-3)
    ✓ window size 10 x 10, view size -4 x -4 located (-4,-4)
    ✓ window size 10 x 10, view size -5 x -5 located (-5,-5)
    ✓ window size 10 x 10, view size -6 x -6 located (-6,-6)
    ✓ window size 10 x 10, view size -7 x -7 located (-7,-7)
    ✓ window size 10 x 10, view size -8 x -8 located (-8,-8)
    ✓ window size 10 x 10, view size -9 x -9 located (-9,-9)
    ✓ window size 10 x 10, view size -10 x -10 located (-10,-10)
```

### Some video for how it looks
<details>
<summary>User has sent tickets, button is visible on the my tickets screen, popover shown once</summary>

https://github.com/AtB-AS/mittatb-app/assets/1777333/fa7275ec-6547-417b-827d-a311e1193b4f

</details>

<details>
<summary>User has no active tickets and purchase tickets for others for the first time, popover shown once</summary>

https://github.com/AtB-AS/mittatb-app/assets/1777333/e2264923-4f2d-420b-b962-21b57f84a0a5

</details>

### CODE REVIEW

**Review by gorandalum — COMMENTED** [review]
I think we should take a look at how we solve the situations where the component which is going to have a popover is not in the viewport. It might be like this, but would like to think it through, and also not merge it right before release since it is kinda complex.

So let's use our time and not rush it.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx
Comment (jorelosorio) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx:86]: Perhaps is better to ask for rectBottom >= 0 to avoid for some strange reason a negative number.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx
Comment (jorelosorio) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx:87]: this rectTop > 0 validations might not be needed.

**Review by jorelosorio — APPROVED** [review]
GW

**Comment by reidzeibel** [comment]
@gorandalum ready for another round of review, I removed the scrolling part but kept the visibility on screen part, because if we don't check for visibility, then the popover would show outside of the screen.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx
Comment (gorandalum) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx:84]: Can create a util function for this code with some unit tests and js-doc.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx
Comment (gorandalum) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_ActiveFareProductsTabScreen.tsx:144]: Adding this `View` breaks the section, and won't give bottom rounding. The `Section` should only contain `SectionItem`s as children.

**Comment by reidzeibel** [comment]
@gorandalum I have committed some of the fix for your comments, the remaining part of this is either :

- how to make the popOverContext accepts `View` Ref **OR**
- should we change the logic of "visibility" for the "Tickets sent to others" button

I will work on the unit test if we decide to go with current solution.

File: src/utils/__tests__/is-view-visible-on-screen.test.ts
Comment (reidzeibel) [inline src/utils/__tests__/is-view-visible-on-screen.test.ts:1]: This is the unit test

File: src/components/sections/items/LinkSectionItem.tsx
Comment (gorandalum) [inline src/components/sections/items/LinkSectionItem.tsx:86]: I have a feeling this ref should be applied to the `PressableOpacity` instead of the `View`. The `PressableOpacity` is the component that is accessible, and most often when we are passing refs it is to focus something for the screen reader.

File: src/utils/is-view-visible-on-screen.ts
Comment (gorandalum) [reply src/utils/is-view-visible-on-screen.ts:1]: Wow! Great documentation!

**Review by gorandalum — APPROVED** [review]
Nice!

File: src/components/sections/items/LinkSectionItem.tsx
Comment (reidzeibel) [reply src/components/sections/items/LinkSectionItem.tsx]: Thanks, I was putting it on the View to test whether the ref works or not, since it works, it should be moved to the `PressableOpacity` 👍

File: src/utils/is-view-visible-on-screen.ts
Comment (marius-at-atb) [inline src/utils/is-view-visible-on-screen.ts:82]: Could consider naming it `isElementFullyInsideScreen`, and also viewX,viewY -> elementX,elementY. 

Reasons:
- to clarify that < 100% visible is not enough to return true
- since this function doesn't know whether the view has opacity: 0, or is hidden behind other elements
- because this function is element agnostic (doesn't require the numbers to belong to a View specifically)

Comment (reidzeibel) [reply src/utils/is-view-visible-on-screen.ts:82]: I agree, was waiting if there are inputs about the naming, so I will rename this in another MR :)

Comment (reidzeibel) [reply src/utils/is-view-visible-on-screen.ts:82]: renamed here : https://github.com/AtB-AS/mittatb-app/pull/4363

**Pull request merged**
